src/helpers/email_helpers.ts

```diff
@@ -0,0 +1,54 @@
+import sendgrid from "@sendgrid/mail";
+import { add_audit_record, get_document_by_id } from ".";
+import { isObject } from "lodash";
+import { EmailSettings, Mail } from "../types";
+import { logger } from "../managers";
+
+export const send_email = async (mail: Mail) => {
+    try {
+        const emails_settings = (await get_document_by_id("nx-settings", "emails")) as EmailSettings;
+        const { sendgrid_api_key, groups, default_from } = emails_settings;
+        let { from, to, cc, group_name, html, text, subject, entity_for_audit } = mail;
+        /// validate data
+        if (from && (typeof from !== "string" || !isObject(from))) {
```

Comment:
will compiler on client side allow to pass here something different from string or object ?